vault/provider.go

```diff
@@ -85,13 +85,16 @@ func Provider() terraform.ResourceProvider {
 
 		DataSourcesMap: map[string]*schema.Resource{
 			"vault_generic_secret": genericSecretDataSource(),
+			"vault_aws_secret":     awsSecretDataSource(),
 		},
 
 		ResourcesMap: map[string]*schema.Resource{
-			"vault_auth_backend":   authBackendResource(),
-			"vault_generic_secret": genericSecretResource(),
-			"vault_policy":         policyResource(),
-			"vault_mount":          mountResource(),
+			"vault_auth_backend":       authBackendResource(),
+			"vault_generic_secret":     genericSecretResource(),
+			"vault_policy":             policyResource(),
+			"vault_mount":              mountResource(),
+			"vault_aws_secret_backend": awsSecretBackendResource(),
+			"vault_aws_secret_role":    awsSecretRoleResource(),
```

Comment:
If this is aligning with some terminology in Vault land then all of this comment is moot and ignorable, but...

"Secret Role" doesn't really _speak to me_ as a what it seems to be here. I think this name would ideally have "credentials" in it to make it clear that what we're issuing here is credentials, rather than just generically a "secret" (which could, for example, be an AWS KMS thing, or an IAM server certificate key, etc).

I took a quick skim of the Vault docs related to this backend, and based on that (it looks like they use the terminology "access credentials" there) my suggestion would be `vault_aws_access_credentials`, to make it very explicit that it's access credentials that we're issuing.

Reply:
So for the line you specifically commented on, the relevant Vault docs say:

> The next step is to configure a role. A role is a logical name that maps to a policy used to generated those credentials. You can either supply a user inline policy (via the policy argument), or provide a reference to an existing AWS policy by supplying the full ARN reference (via the arn argument).
>
> For example, lets first create a "deploy" role using an user inline policy as an example:

So I think `vault_aws_access_credentials` doesn't quite capture the idea of it, if that's what you were suggesting get renamed. Though, looking at the pattern established in later resources, I renamed it `vault_aws_secret_backend_role` instead of `vault_aws_secret_role`, which feels a bit clearer.

Similarly, for `vault_aws_secret_backend`, the docs say:

> The first step to using the aws backend is to mount it. Unlike the kv backend, the aws backend is not mounted by default.
>
> {code_snippet}
>
> Next, we must configure the credentials that Vault uses to manage the IAM credentials generated by this secret backend:

This is encapsulated in the `vault_aws_secret_backend` resource. I didn't call it `vault_aws_secret_backend_credentials` because it takes care of both the mounting _and_ the credentials configuration, so it seemed clearer to have the name imply we were configuring the entire backend, not just the credentials. The reason the two API calls are combined like that is because removing the credentials once they're configured is impossible, and setting them to an invalid value breaks things, so I was struggling with the Delete function. The canonical way to remove configured credentials, as far as I can tell, is to unmount the backend. So I combined mounting/unmounting the backend and configuring the credentials so it all maps nicely to the CRUD format Terraform expects.

All that said, I can see an argument for renaming the `vault_aws_secret` data source to `vault_aws_access_credentials`, if that's what you're referencing. The `_secret` part largely just came from trying to match the pre-existing `generic_secret`, to make it obvious that that's generic and this is the aws-specific flavor. But on further consideration, I _do_ like `vault_aws_access_credentials` better, so I've updated the name to reflect that. If any of this is unclear or does not make sense, or you disagree with it, do let me know.